luaui/Widgets/export_defs.lua

```diff
@@ -0,0 +1,60 @@
+function widget:GetInfo()
+	return {
+		name = "Unitdefs JSON Export",
+		desc = "Exports UnitDefs into JSON files.",
+		author = "Wereii",
+		date = "June 2023",
+		license = "GNU GPL, v2 or later, BA/BAR only",
+		layer = 99999,
+		enabled = false
+	}
+end
+
+local export_folder_path = "json_export/"
+
+local function TableToFile(tbl, filename)
+	local file, err = io.open(filename, "w")
+	if not file then
+		Spring.Echo("Error opening file: " .. err)
+		return
+	end
+	file:write(Json.encode(tbl))
+	file:close()
+end
+
+local function ExportDefs()
+	if not VFS.FileExists(export_folder_path) then
+		Spring.CreateDir(export_folder_path)
+	end
+
+	for id, unitDef in pairs(UnitDefs) do
+		Spring.Echo(string.format("Exporting unitdef: %s", unitDef.name))
+		local tbl = {}
+
+		tbl["id"] = unitDef.id
+		-- embed higher-level "computed" fields like translatedHumanName, translatedTooltip, etc.
+		for field_name, value in pairs(unitDef) do
+			if type(value) ~= "table" or type(value) ~= "function" or type(value) ~= "nil" then
+				tbl[field_name] = value
+			end
+		end
+
+		-- flatten the UnitDef metatable data into plain table
+		for k, v in unitDef:pairs() do
+			tbl[k] = v
+		end
+
+		TableToFile(tbl, string.format("json_export/%s.json", unitDef.name))
+	end
+end
+
+function widget:TextCommand(command)
+	if string.find(command, "exportdefs", nil, true) == 1 and string.len(command) == 10 then
```

Comment:
Why not just string equality when you search for string of length 10 and then check that whole lenght is 10?

Reply:
Checks that this is exactly `/exportdefs` not `/exportdefs that`, `/exportdefsssss`.
(Also copied from other widget :D)

Reply:
I don't understand why it's not just a full string equality but maybe I again just don't know something about lua

Reply:
probably copy-pasted from other widgets that accept other args? doing `command == "exportdefs"` would make sense, but if you've already got this check then perhaps consider accepting a parameter to only dump 1 (or N) defs like `/exportdefs armpw` or `/exportdefs armrock armjeth`

Reply:
>  widgets that accept other args 

Oh right, that's probably it. 

> accepting a parameter to only dump 1 (or N) defs

Could do, but is it worth making the widget more complex like this? I guess one use of that would be when changing units to see only one or few quickly without having to export all (itl takes about 1 second for all).